iot_config/src/lora_field.rs

```diff
@@ -0,0 +1,516 @@
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{fmt::Display, str::FromStr};
+
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
+pub struct LoraField<const WIDTH: usize>(pub u64);
```

Comment:
A quick scan through this code doesn't make it obvious what this is for, but I'm guessing it has to do with parsing?

Reply:
correct; this is for bit twiddling NetIDs, DevAddrs, and Euis; just below that i implement each of these:
```
pub type NetIdField = LoraField<6>;
pub type DevAddrField = LoraField<8>;
pub type EuiField = LoraField<16>;
```
and then further down have impl blocks that add specific behavior on those three type. this is liberally stolen from work @michaeldjeffrey did in the rust CLI